src/utils/helpers/source.helper.ts

```diff
@@ -224,47 +229,56 @@ export class SourceHelper {
    *
    * @param id The ID of the player
    */
-  static hook(id: string): void {
-    const elementId = `#player-${id}`;
-
-    DOMHelper
-      .watch(elementId)
-      .then(e => e[0] as HTMLVideoElement)
-      .then(player => {
-        const store = useSourcesStore();
+  static async hook(id: string): Promise<void> {
+    return new Promise(resolve => {
+      const elementId = `#player-${id}`;
+
+      DOMHelper
+        .watch(elementId)
+        .then(e => e[0] as HTMLVideoElement)
+        .then(player => {
+          const store = useSourcesStore();
+          const source = store.getSource(id);
+
+          player.volume = source.metadata.volume;
```

Comment:
Something's wrong here!!

Reply:
Fixed by the latest commit. [7571f08](https://github.com/EOussama/playback-sync/pull/139/commits/7571f0812b5dec36fcee8beee8a0d6cd174eafb9).